addon/collapse-key.js

```diff
@@ -5,11 +5,15 @@ export default function(property) {
     return [property];
   }
 
-  let atEachIndex = property.indexOf('.@each');
-  if (atEachIndex !== -1) {
-    return [property.slice(0, atEachIndex)];
-  } else if (property.slice(-2) === '[]') {
-    return [property.slice(0, -3)];
+  let atEachIndex = property.indexOf('@each.');
+  if (atEachIndex === -1) {
+    atEachIndex = property.indexOf('[]');
+  }
+
+  if (atEachIndex === 0) {
+    return [''];
```

Comment:
I tried this branch out, and I think you want to `return [];` in this case. Then you can remove the extra blank handling from `get-value.js`.